src/iidm/converter/xml/IidmXmlUtil.cpp

```diff
@@ -21,38 +26,68 @@ namespace converter {
 
 namespace xml {
 
+std::string message(const std::string& elementName, const char* errorMessage, const IidmXmlVersion& version, const IidmXmlVersion& contextVersion, const std::string& reason) {
+    return stdcxx::format("%1% is %2% for IIDM-XML version %3%. %4%%5%", elementName, errorMessage, contextVersion.toString("."), reason, version.toString("."));
+}
+
+PowsyblException createException(const std::string& elementName, const char* errorMessage, const IidmXmlVersion& version, const IidmXmlVersion& contextVersion, const std::string& reason) {
+    return PowsyblException(message(elementName, errorMessage, version, contextVersion, reason));
+}
+
 PowsyblException createException(const std::string& rootElementName, const std::string& elementName, const char* errorMessage, const IidmXmlVersion& version, const IidmXmlVersion& contextVersion, const std::string& reason) {
-    return PowsyblException(stdcxx::format("%1%.%2% is %3% for IIDM-XML version %4%. %5%%6%", rootElementName, elementName, errorMessage, contextVersion.toString("."), reason, version.toString(".")));
+    return createException(stdcxx::format("%1%.%2%", rootElementName, elementName), errorMessage, version, contextVersion, reason);
 }
 
-void IidmXmlUtil::assertMaximumVersion(const std::string& rootElementName, const std::string& elementName, const char* errorMessage, const IidmXmlVersion& maxVersion, const IidmXmlVersion& contextVersion) {
-    if (contextVersion > maxVersion) {
-        throw createException(rootElementName, elementName, errorMessage, maxVersion, contextVersion, "IIDM-XML version should be <= ");
+void createExceptionOrLogError(const std::string& elementName, const char* errorMessage, const IidmXmlVersion& refVersion, const std::string& reason, const NetworkXmlWriterContext& context) {
+    static logging::Logger& S_LOGGER = logging::LoggerFactory::getLogger<IidmXmlUtil>();
+    if (context.getOptions().getIidmVersionIncompatibilityBehavior() == ExportOptions::IidmVersionIncompatibilityBehavior::THROW_EXCEPTION) {
+        throw createException(elementName, errorMessage, refVersion, context.getVersion(), reason);
     }
+    if (context.getOptions().getIidmVersionIncompatibilityBehavior() == ExportOptions::IidmVersionIncompatibilityBehavior::LOG_ERROR) {
+        S_LOGGER.error(message(elementName, errorMessage, refVersion, context.getVersion(), reason));
```

Comment:
I think it's not a good idea to store the logger in a static variable in the C++ implementation because it's possible to add/remove logger after the startup of the application. Maybe a local variable declare in this scope is sufficient.

Regarding to the coding style, the `S_LOGGER` should be written in lower case.